File: src/types/character.ts
import type { TTSConfig } from './tts';

/** AIキャラクター */
export interface Character {
  id: string;
  name: string;
  /** ユーザーが入力した概要説明 */
  description: string;
  /** LLM生成 or 手動編集されたシステムプロンプト */
  system_prompt: string;
  avatar_path?: string;
  tts_config?: TTSConfig;
  /** ISO 8601 */
  created_at: string;
  /** ISO 8601 */
  updated_at: string;
}

/** キャラクター更新用（部分更新対応） */
export interface CharacterUpdate {
  name?: string;
  description?: string;
  system_prompt?: string;
  avatar_path?: string;
  tts_config?: TTSConfig;
  /** trueの場合、avatar_pathをNULLに更新する */
  clear_avatar?: boolean;
  /** trueの場合、tts_configをNULLに更新する */
  clear_tts?: boolean;
}

File: src-tauri/src/models/character.rs
use serde::{Deserialize, Serialize};

use super::tts::TTSConfig;

/// AIキャラクター
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Character {
    /// UUID v4
    pub id: String,
    pub name: String,
    /// ユーザーが入力した概要説明
    pub description: String,
    /// LLM生成 or 手動編集されたシステムプロンプト
    pub system_prompt: String,
    pub avatar_path: Option<String>,
    pub tts_config: Option<TTSConfig>,
    /// ISO 8601
    pub created_at: String,
    /// ISO 8601
    pub updated_at: String,
}

/// キャラクター更新用（部分更新対応）
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CharacterUpdate {
    pub name: Option<String>,
    pub description: Option<String>,
    pub system_prompt: Option<String>,
    pub avatar_path: Option<String>,
    pub tts_config: Option<TTSConfig>,
    /// trueの場合、avatar_pathをNULLに更新する
    pub clear_avatar: Option<bool>,
    /// trueの場合、tts_configをNULLに更新する
    pub clear_tts: Option<bool>,
}
